File: 1922-count-good-numbers/1922-count-good-numbers.java
class Solution {
    private long mod = 1_000_000_007;
    public int countGoodNumbers(long n) {
        long e = (n+1)/2;
        long o = n/2;
        long f = pow(5,e)%mod;
        long s = pow(4,o)%mod;
        return (int)((f*s)%mod);
    }
    private long pow(long x, long n){
        if(n==0) return 1;
        long temp = pow(x,n/2);
        if(n%2==0){
            return (temp*temp)%mod;
        }else{
            return (x*temp*temp)%mod;
        }
    }
}

File: 1922-count-good-numbers/1922-count-good-numbers.py
class Solution:
    mod = 10**9+7
    def power(self,x,y):
        if y==0:
            return 1
        ans = self.power(x,y//2)
        ans *= ans
        ans %= self.mod
        if y%2!=0:
            ans *= x
            ans %= self.mod
        return ans
    def countGoodNumbers(self, n: int) -> int:
        e = (n//2+n%2)
        o = n//2
        return (self.power(5,e)*self.power(4,o))%self.mod
        